File: sdk/csharp/src/Buildkite.Sdk/TriggerStep.cs
using System.Collections.Generic;

namespace Buildkite.Sdk
{
    public class TriggerStep : PipelineStep
    {
        public string? Trigger { get; set; }
        public Build? Build { get; set; }
        public bool? Async { get; set; }

        public TriggerStep()
        {
        }

        public TriggerStep(string trigger)
        {
            Trigger = trigger;
        }

        public TriggerStep(string label, string trigger)
        {
            Label = label;
            Trigger = trigger;
        }
    }
}

File: sdk/typescript/src/triggerStep.ts
import { Build, DependsOn, SoftFail } from './types'

export interface TriggerStep {
    allow_dependency_failure?: boolean;
    async?: boolean;
    branches?: string[] | string;
    build?: Build;
    depends_on?: Array<DependsOn | string> | null | string;
    id?: string;
    identifier?: string;
    if?: string;
    key?: string;
    label?: string;
    name?: string;
    skip?: boolean | string;
    soft_fail?: SoftFail[] | boolean;
    trigger: string;
}
